Eastern-Research-Group/expert-query
# Feature/263 download modal changes

## Related Issues:
* [EQ-263](https://jira.epa.gov/browse/EQ-263)
* [EQ-264](https://jira.epa.gov/browse/EQ-264)
* [EQ-265](https://jira.epa.gov/browse/EQ-265)
* [EQ-271](https://jira.epa.gov/browse/EQ-271)
* [EQ-273](https://jira.epa.gov/browse/EQ-273)

## Main Changes:
* Added a loading spinner icon to the "Continue" button in the download modal, which shows while the query is executing.
  * It still changes to show "Working", since the button becomes disabled and it felt odd to still show "Continue".
* Added a call to focus the Cancel button of the download modal as soon as the count query finishes, to align with the clear search modal behavior.
* Added abort signals to the queries and count queries, so the request is aborted when the modal is closed. The query still executes on the back-end, but this avoids some network traffic and stops the download window from unexpectedly opening later.
* Removed the glossary panel.
* Updated the "Advanced API Queries" accordion heading to "Advanced Queries" (though we'll be changing this soon).
* Fixed some button alignment (_Clear Search_, _Download_, _Continue_).
* Fixed the issue of the `locationTypeCode` source select drop-down causing a scrollbar to appear in the Assessment Units profile.
* Removed the timeout on the success message that displays when a download completes, and added to the message.

## Steps To Test:
1. Check the style changes described above. Ensure they are consistent across screen sizes.
2. Open the console's network panel, and attempt to execute a query on the whole dataset (it might help to be pointed at the dev database).
3. While the count query is executing, close the modal. The count should be aborted. _N.B._ I did verify that the duplicate count request is simply due to React's strict mode.
4. Repeat 3, but wait until the count is complete and close the modal after clicking "Continue".
